src/chm/documents/bundle/bundle_built_in_variables.html.md

```diff
@@ -24,6 +24,7 @@ The Burn engine offers a set of commonly-used variables so you can use them with
 * LocalAppDataFolder - gets the well-known folder for CSIDL\_LOCAL\_APPDATA.
 * LogonUser - gets the current user name.
 * MyPicturesFolder - gets the well-known folder for CSIDL\_MYPICTURES.
+* NativeMachine - gets the [Image File Machine value](https://docs.microsoft.com/en-us/windows/win32/sysinfo/image-file-machine-constants) representing the native architecture of the machine. This property is only set on Windows 10, version 1511 (TH2) and higher.
```

Comment:
I don't like the name `NativeMachine`, I think that only makes sense if you're looking at the `IsWow64Process2` documentation. Perhaps `NativeProcessorArchitecture`? Either way, probably should remove the word "native" from the documentation for `ProcessorArchitecture`.

Reply:
It is tied to “image file machine constants” as well https://docs.microsoft.com/en-us/windows/win32/sysinfo/image-file-machine-constants.  You can’t possibly use this property unless you read those docs due to the esoteric number values.

If we call it something different we risk making some sort of promise other than what the underlying API is making. I prefer to make this a direct pass through to that underlying API, naming and all.

Reply:
To be clear, I'm happy to change this to whatever folks decide, but wanted to share my reasoning for the current naming.  @barnson @robmen can you let me know what to do to establish consensus on this?  I'm happy to write something up if necessary.

Reply:
I'm OK with a "pass-through" value so we don't have to make judgment calls about which platforms we should "promote." That said, when I made the original pass for ARM64/x64 support in Burn, I added both the Burn platform and native platform to `EngineRun`. (Though I did so using `GetNativeSystemInfo`. 💢) Having the platforms available as strings would make using them a lot prettier. I'd probably split the difference and say that v3 is good enough with inscrutable numbers and at some point in v4.0/v4.x, we should refactor the extant code to get strings in a new variable.

Reply:
Once you add string translations to this you make a promise that you'll update it once some new platform comes along and needs a string value.  If you just pass the integer through then folks can depend on whatever the Windows API returns, even if new values show up in the future.  IMHO that's better then the minor convenience of a string.  Folks will likely define wix variables or defines with expected integer values anyway.  What's your thinking about adding something like this to the extension to make that easier?
```XML
<!-- IMAGE_FILE_MACHINE_UNKNOWN 0 -->
<WixVariable Name"FileMachineUnknown" Value="0" />
<!-- IMAGE_FILE_MACHINE_I386 0x014c (332) -->
<WixVariable Name"FileMachineI386" Value="332" />
<!-- IMAGE_FILE_MACHINE_AMD64 0x8664 (34404) -->
<WixVariable Name"FileMachineAMD64" Value="34404" />
<!-- IMAGE_FILE_MACHINE_ARM64 0xAA64 (43602) -->
<WixVariable Name"FileMachineARM64" Value="43602" />
```
That way folks aren't blocked if a new platform comes along, and we could document these as part of the extension.

Reply:
Providing "constants" is a decent workaround, though they'd have to be namespaced/reserved. OTOH, I'm not exactly worried about a new architecture just popping up out of nowhere.

Reply:
> I'm not exactly worried about a new architecture just popping up out of nowhere.

Look at this case.  We have 5 year old servicing branches that didn't anticipate this but needed to have installer work done to play nicely with it, without explicitly supporting it.  It would be nice if we didn't need to take toolchain updates to enable that.

Reply:
> It would be nice if we didn't need to take toolchain updates to enable that.

Meh. It's not a bad goal but the probability of that happening is low. You might remember that .NET didn't have any ARM64 installers until after I did the work to support ARM64 in WiX. (You're totally welcome! No thanks needed!) That there's also a tweak required for x86-on-ARM64 is...unsurprising.

But to be clear, I'm not opposed to exposing the "raw" values.

Reply:
> It's not a bad goal but the probability of that happening is low

Agreed.

> You're totally welcome! No thanks needed!

Thank you indeed, I wasn't aware of that.  Been away from installer space for a bit, but I'm happy to come back to help.  This codebase brings back lots of fun memories.  What changes do you guys think are necessary to get this and the wix4 PR in?  Also I plan to attend Thursday's meeting if that's a better place to discuss.

Reply:
https://www.youtube.com/watch?v=UPw-3e_pzqU

Thursday's meeting is probably where the discussion should happen, as it's a change in v3.x. I'll take a look at the v4 pull request before then.